### ISSUE TITLE
Yolov8 Python SDK crash after closing dataloader mosaic 

### ISSUE THREAD

**Issue by nghiakthp2401** [ultralytics/ultralytics#14930]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

![image](https://github.com/user-attachments/assets/4af9a5cb-7c60-4511-a72d-9fb32818f058)

My code: 

''''
    weightPath = "runs\\detect\\train9\\weights\\\last.pt"
    datasetPath = "D:\\Nghia\\CustomYolov8\\YoloV8-Custom\\TrainFolder\\data.yaml"

    ModifyDataYamlFile(datasetPath, False, True)

    #Tranning Pharse
    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
    model = YOLO(weightPath)
    model.add_callback("on_train_start", tensorboard_Start)

    print(type(model))
    results = model.train(data=datasetPath, epochs=2000, batch = 64, 
                    patience= 0, imgsz = 640,  device=device, workers= 1,
                     flipud=0, fliplr=0, mosaic = 0, lr0=0.0001, optimizer="SGD"
                   )
''''

### Environment

Ultralytics YOLOv8.2.70 🚀 Python-3.9.13 torch-1.13.1+cu117 CUDA:0 (NVIDIA GeForce RTX 4070 Ti, 12282MiB)

OS                  Windows-10-10.0.19045-SP0
Environment         Windows
Python              3.9.13
Install             pip
RAM                 63.78 GB
CPU                 12th Gen Intel Core(TM) i7-12700
CUDA                11.7

numpy               ✅ 1.26.4<2.0.0,>=1.23.0
matplotlib          ✅ 3.9.1>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.13.1>=1.4.1
torch               ✅ 1.13.1+cu117>=1.8.0
torchvision         ✅ 0.14.1+cu117>=0.9.0
tqdm                ✅ 4.66.4>=4.64.0
psutil              ✅ 6.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.0>=2.0.0

### Minimal Reproducible Example

'''

if __name__ == '__main__':

    #Robotflow Data.yaml path
    weightPath = "runs\\detect\\train9\\weights\\\last.pt"
    datasetPath = "D:\\Nghia\\CustomYolov8\\YoloV8-Custom\\TrainFolder\\data.yaml"

    ModifyDataYamlFile(datasetPath, False, True)

    #Tranning Pharse
    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
    model = YOLO(weightPath)
    model.add_callback("on_train_start", tensorboard_Start)

    print(type(model))
    results = model.train(data=datasetPath, epochs=2000, batch = 64, 
                    patience= 0, imgsz = 640,  device=device, workers= 1,
                     flipud=0, fliplr=0, mosaic = 0, lr0=0.0001, optimizer="SGD"
                   )
'''

### Additional

Some other Function in code but not relative to that bug:
'''
from ultralytics import YOLO
from os import path, getcwd, listdir
from tensorboard import program
import torch

#Change Loss Type 

def ModifyDataYamlFile(filePath, UseTrainAsVal, defaultData = True):

    if( defaultData):
        StartUpPath = getcwd()
        with open(filePath, "r") as file:
            lines = file.readlines()
            lines[0] = "train: " + StartUpPath + "\\YoloV8-Custom\\TrainFolder\\train\\images\n"
            if UseTrainAsVal:
                lines[1] = "val: " + StartUpPath + "\YoloV8-Custom\\TrainFolder\\train\\images\n"
            else :
                lines[1] = "val: " + StartUpPath + "\YoloV8-Custom\\TrainFolder\\valid\\images\n"
            lines[2] = "test: " + StartUpPath + "\YoloV8-Custom\\TrainFolder\\test\\images\n"
        with open(filePath, "w") as file:
            file.writelines(lines)
    else:
        with open(filePath, "r") as file:
            lines = file.readlines()
            lines[0] = "train: " + '\\'.join(filePath.split('\\')[0:-1]) + "train\\images\n"
            if UseTrainAsVal:
                lines[1] = "val: " + '\\'.join(filePath.split('\\')[0:-1]) + "train\\images\n"
            if UseTrainAsVal:
                lines[1] = "val: " + '\\'.join(filePath.split('\\')[0:-1]) + "train\\images\n"
            lines[2] = "test: " + '\\'.join(filePath.split('\\')[0:-1]) + "train\\images\n"

def GetResultLogDir():
    arraySectionTrain = []
    print(getcwd())
    
    runPath = path.join(path.join(getcwd(), "runs\\detect"))
    print(runPath)
    if(path.exists(runPath)):
        for filename in listdir(runPath):
            try:
                arraySectionTrain.append((int)(filename.split('train')[1]))
            except:
                None
        if(len( arraySectionTrain) == 0): return path.join(runPath, "train")
        a = max(arraySectionTrain)
        return path.join(runPath, "train" + (str)(max(arraySectionTrain)))
    return path.join(runPath, "train")

def tensorboard_Start(trainer):
    log_dir = GetResultLogDir()
    reload_interval = 30
    tb = program.TensorBoard()
    tb.configure(argv=[None, '--logdir', log_dir, '--reload_interval', str(reload_interval)])
    url = tb.launch()
    print(f"TensorBoard is running at {url}")

'''

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by Y-T-G** [comment]
Seems like a syntax error in your modifications. Not ultralytics related.

**Comment by nghiakthp2401** [comment]
> Seems like a syntax error in your modifications. Not ultralytics related.

I think 

> Seems like a syntax error in your modifications. Not ultralytics related.

i installed ultralystic in virtualenv then that's error happens. Run without virtualenv stills OK

**Comment by pderrenger** [comment]
@nghiakthp2401 thank you for the additional information. To help us pinpoint the issue, could you please ensure that you are using the latest version of the Ultralytics package? Sometimes, issues in virtual environments can be resolved by updating dependencies. If the problem persists, please let us know if there are any specific error messages or logs that could provide more insight.

**Comment by nghiakthp2401** [comment]
> @nghiakthp2401 thank you for the additional information. To help us pinpoint the issue, could you please ensure that you are using the latest version of the Ultralytics package? Sometimes, issues in virtual environments can be resolved by updating dependencies. If the problem persists, please let us know if there are any specific error messages or logs that could provide more insight.

i used ultralytics-thop ✅ 2.0.0>=2.0.0

**Comment by pderrenger** [comment]
Thank you for confirming the version. Please ensure that all dependencies, including PyTorch, are up-to-date. If the issue persists, could you provide any specific error messages or logs? This will help us diagnose the problem more effectively.
